fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -184,48 +218,24 @@ where
             return Err(Error::ParentChainReorgDetected);
         }
 
-        if let Some((to_confirm_height, to_confirm_hash)) = self.sync_pointers.tail() {
-            tracing::debug!(
+        let data = self.fetch_data(height, block_hash_res.block_hash).await?;
+        atomically_or_err::<_, Error, _>(|| {
+            self.provider.new_parent_view(height, Some(data.clone()))?;
+            emit!(
```

Comment:
This too should be outside the `atomically_or_err`.

Reply:
location shifted